packages/headless/src/features/search-parameters/search-parameter-serializer.ts

```diff
@@ -15,7 +15,7 @@ function serialize(obj: SearchParameters) {
 function deserialize(fragment: string): SearchParameters {
   const parts = fragment.split('&').filter((part) => part.length);
```

Comment:
So the deserializer will correctly parse that string, however, it will incorrectly parse a string where the `&` is used as a value. e.g.

```
q=a&b&aq=@author==alice

//should give
{q: 'a&b', aq: '@author==alice'}
```

I think we should encode the character if it is found in the value. I will open a jira to explore this separately,